### ISSUE TITLE
calculating memory requirements based on bam file size

### ISSUE THREAD

**Issue by avilella** [biocore-ntnu/epic#86]
Is it possible to approximately calculate the memory requirements of EPIC based on the size or number of reads of the bam files given as input? This is in line with the efforts to try to give the smallest possible instance to the job that will not run out of memory.

**Comment by endrebak** [comment]
No, I have not thought much about memory-requirements. Just speed. But epic is memory-intensive, which is what allows it to create all those nice bigwigs/matrixes in the end.

**Comment by endrebak** [comment]
Having two scripts, one to produce the enriched regions and another one (using the original input files and the enriched regions) to produce bigwigs and matrixes would make the first much more memory efficient and even faster. But as I do not see a paper coming out of epic, I do not have the resources to prioritize it :/

**Comment by avilella** [comment]
I see. Do you mean that the two script option would work now or that it
would require some code to get working?

**Comment by endrebak** [comment]
Sorry for being unclear. It was almost halfway a note to self. I could create an epic-light that does not preserve any bin-info other than that required to find the enriched regions. But I do not see that coming out anytime soon due to resource constraints.

**Comment by endrebak** [comment]
I have realized that in order to get epic accepted as an application note I should reduce the memory requirements greatly. This is not possible for the bigwig/matrix-producing parts, but should be possible when just calling islands. My first priority is getting pyranges out there though, since I see it  as a foundational library for genomics/bioinformatics, while epic is just a piece of software for a very specific task. It is in my backlog though.

Unfortunately I cannot _guarantee_ that it will ever happen :/

**Comment by endrebak** [comment]
@avilella See SICER2 at https://github.com/endrebak/SICER2

![memory_sicer2_vs_sicer_no_bigwig](https://user-images.githubusercontent.com/17778898/47266140-a8179900-d532-11e8-9430-b2e805c7f04b.png)
![speed_sicer2_vs_sicer_no_bigwig](https://user-images.githubusercontent.com/17778898/47266146-b796e200-d532-11e8-876d-049c7b6ae220.png)
